x-pack/plugins/lens/public/heatmap_visualization/suggestions.ts

```diff
@@ -22,7 +22,9 @@ export const getSuggestions: Visualization<HeatmapVisualizationState>['getSugges
     (state?.shape === CHART_SHAPES.HEATMAP &&
       (state.xAccessor || state.yAccessor || state.valueAccessor) &&
       table.changeType !== 'extended') ||
-    table.columns.some((col) => col.operation.isStaticValue)
+    table.columns.some((col) => col.operation.isStaticValue) ||
+    // do not use suggestions with non-numeric metrics
+    table.columns.some((col) => !col.operation.isBucketed && col.operation.dataType !== 'number')
```

Comment:
I see we have a duplication `(col.operation.isStaticValue ||  !col.operation.isBucketed && col.operation.dataType !== 'number' )`  of that code, maybe it's possible to unify it somehow?

creating a simple util method can help us further maintain this code.